src/middle/Compiler_intf.ml

```diff
@@ -0,0 +1,24 @@
+open Core_kernel
+
+module type S = sig
+  type frontend_error
+  type compiler_opts_error
+  type compiler_opts
+
+  val default_compiler_opts : compiler_opts
+
+  val compiler_opts_of_string :
+    string -> (compiler_opts, compiler_opts_error list) result
+
+  val compile_from_file :
+    opts:compiler_opts -> file:string -> (string, frontend_error) result
```

Comment:
Is this at all related to the PR? Also, just wondering: does it matter whether this is a .ml or .mli file?

Reply:
Same as above regarding; it's in an .ml file since we want to expose the module signatures and functor from middle so that it can be used to either write new frontends/backends or construct the core compiler with them

Reply:
Could you not do that if it's in a .mli file? I guess I still don't fully understand when you put signatures in a .mli file and when in a .ml file.